src/modules/graphql/resolvers/student.resolver.ts

```diff
@@ -1,46 +1,95 @@
 import { Resolver, Query, Mutation, ResolveProperty } from '@nestjs/graphql';
 import { UsersPersistenceService } from '../../persistence/users.persistence.service';
 import { ClassPersistenceService } from '../../persistence/class.persistence.service';
-import { UserRole } from '../../../models/user.db.model';
+import { UserDbModel, UserRole} from '../../../models/user.db.model';
+import { Asset, DBOperation, NO_PERMISSION, Permission, Permissions} from './permissionRules';
+import { Get } from '../../../utils/get';
 
 @Resolver('Student')
 export class StudentResolver {
     constructor(private usersPersistence: UsersPersistenceService, private classPersistence: ClassPersistenceService) { }
 
     @Query('students')
-    async getStudents() {
+    async getStudents(_, {}, context) {
+        const permission = Permissions.checkAndGetBasePermission(Get.getObject(context, 'user'), DBOperation.READ, Asset.STUDENT);
```

Comment:
same here, maybe if this scenario (do something for some permission) occurs often maybe some sort of strategy pattern can be generated

Reply:
added strategies + willbe refactoredwith new service